CloudFront Distributions Failing to Delete on master

After upgrading to aws-sdk-go v1.13.28 in #4048, the acceptance testing is now failing for all CloudFront distribution tests with:

```
=== RUN   TestAccAWSCloudFrontDistribution_HTTP11Config
--- FAIL: TestAccAWSCloudFrontDistribution_HTTP11Config (8.79s)
    testing.go:579: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: Error applying: 1 error(s) occurred:
        
        * aws_cloudfront_distribution.http_1_1 (destroy): 1 error(s) occurred:
        
        * aws_cloudfront_distribution.http_1_1: InvalidArgument: The parameter Field level encryption id is required.
            status code: 400, request id: 5d887e11-3960-11e8-9b3f-e9be4997ebb0
```

### Terraform Version

* Pre-v1.14.0 of the AWS provider

### Affected Resource(s)

* aws_cloudfront_distribution

--- a/aws/cloudfront_distribution_configuration_structure.go
+++ b/aws/cloudfront_distribution_configuration_structure.go
@@ -204,6 +204,9 @@ func defaultCacheBehaviorHash(v interface{}) int {
 	buf.WriteString(fmt.Sprintf("%s-", m["target_origin_id"].(string)))
 	buf.WriteString(fmt.Sprintf("%d-", forwardedValuesHash(m["forwarded_values"].(*schema.Set).List()[0].(map[string]interface{}))))
 	buf.WriteString(fmt.Sprintf("%d-", m["min_ttl"].(int)))
+	if d, ok := m["field_level_encryption_id"]; ok && d.(string) != "" {
+		buf.WriteString(fmt.Sprintf("%s-", d.(string)))
+	}
 	if d, ok := m["trusted_signers"]; ok {
 		for _, e := range sortInterfaceSlice(d.([]interface{})) {
 			buf.WriteString(fmt.Sprintf("%s-", e.(string)))
@@ -266,13 +269,14 @@ func flattenCacheBehaviors(cbs *cloudfront.CacheBehaviors) *schema.Set {
 
 func expandCacheBehavior(m map[string]interface{}) *cloudfront.CacheBehavior {
 	cb := &cloudfront.CacheBehavior{
-		Compress:             aws.Bool(m["compress"].(bool)),
-		ViewerProtocolPolicy: aws.String(m["viewer_protocol_policy"].(string)),
-		TargetOriginId:       aws.String(m["target_origin_id"].(string)),
-		ForwardedValues:      expandForwardedValues(m["forwarded_values"].(*schema.Set).List()[0].(map[string]interface{})),
-		DefaultTTL:           aws.Int64(int64(m["default_ttl"].(int))),
-		MaxTTL:               aws.Int64(int64(m["max_ttl"].(int))),
-		MinTTL:               aws.Int64(int64(m["min_ttl"].(int))),
+		Compress:               aws.Bool(m["compress"].(bool)),
+		FieldLevelEncryptionId: aws.String(m["field_level_encryption_id"].(string)),
+		ViewerProtocolPolicy:   aws.String(m["viewer_protocol_policy"].(string)),
+		TargetOriginId:         aws.String(m["target_origin_id"].(string)),
+		ForwardedValues:        expandForwardedValues(m["forwarded_values"].(*schema.Set).List()[0].(map[string]interface{})),
+		DefaultTTL:             aws.Int64(int64(m["default_ttl"].(int))),
+		MaxTTL:                 aws.Int64(int64(m["max_ttl"].(int))),
+		MinTTL:                 aws.Int64(int64(m["min_ttl"].(int))),
 	}
 
 	if v, ok := m["trusted_signers"]; ok {
@@ -304,6 +308,7 @@ func flattenCacheBehavior(cb *cloudfront.CacheBehavior) map[string]interface{} {
 	m := make(map[string]interface{})
 
 	m["compress"] = *cb.Compress
+	m["field_level_encryption_id"] = aws.StringValue(cb.FieldLevelEncryptionId)
 	m["viewer_protocol_policy"] = *cb.ViewerProtocolPolicy
 	m["target_origin_id"] = *cb.TargetOriginId
 	m["forwarded_values"] = schema.NewSet(forwardedValuesHash, []interface{}{flattenForwardedValues(cb.ForwardedValues)})
@@ -346,6 +351,9 @@ func cacheBehaviorHash(v interface{}) int {
 	buf.WriteString(fmt.Sprintf("%s-", m["target_origin_id"].(string)))
 	buf.WriteString(fmt.Sprintf("%d-", forwardedValuesHash(m["forwarded_values"].(*schema.Set).List()[0].(map[string]interface{}))))
 	buf.WriteString(fmt.Sprintf("%d-", m["min_ttl"].(int)))
+	if d, ok := m["field_level_encryption_id"]; ok && d.(string) != "" {
+		buf.WriteString(fmt.Sprintf("%s-", d.(string)))
+	}
 	if d, ok := m["trusted_signers"]; ok {
 		for _, e := range sortInterfaceSlice(d.([]interface{})) {
 			buf.WriteString(fmt.Sprintf("%s-", e.(string)))

--- a/aws/cloudfront_distribution_configuration_structure_test.go
+++ b/aws/cloudfront_distribution_configuration_structure_test.go
@@ -23,6 +23,7 @@ func defaultCacheBehaviorConf() map[string]interface{} {
 		"allowed_methods":             allowedMethodsConf(),
 		"cached_methods":              cachedMethodsConf(),
 		"compress":                    true,
+		"field_level_encryption_id":   "",
 	}
 }
 

--- a/aws/resource_aws_cloudfront_distribution.go
+++ b/aws/resource_aws_cloudfront_distribution.go
@@ -61,6 +61,10 @@ func resourceAwsCloudFrontDistribution() *schema.Resource {
 							Optional: true,
 							Default:  86400,
 						},
+						"field_level_encryption_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
 						"forwarded_values": {
 							Type:     schema.TypeSet,
 							Required: true,
@@ -211,6 +215,10 @@ func resourceAwsCloudFrontDistribution() *schema.Resource {
 							Type:     schema.TypeInt,
 							Optional: true,
 							Default:  86400,
+						},
+						"field_level_encryption_id": {
+							Type:     schema.TypeString,
+							Optional: true,
 						},
 						"forwarded_values": {
 							Type:     schema.TypeSet,

--- a/website/docs/r/cloudfront_distribution.html.markdown
+++ b/website/docs/r/cloudfront_distribution.html.markdown
@@ -172,6 +172,8 @@ of several sub-resources - these resources are laid out below.
     in the absence of an `Cache-Control max-age` or `Expires` header. Defaults to
     1 day.
 
+  * `field_level_encryption_id` (Optional) - Field level encryption configuration ID
+
   * `forwarded_values` (Required) - The [forwarded values configuration](#forwarded-values-arguments) that specifies how CloudFront
     handles query strings, cookies and headers (maximum one).
 

